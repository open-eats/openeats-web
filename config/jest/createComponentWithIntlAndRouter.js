import React from 'react';
import renderer from 'react-test-renderer';
import { IntlProvider } from 'react-intl';
import { MemoryRouter } from 'react-router-dom'

const createComponentWithIntl = (children, props = { locale: 'en' }) => {
  return renderer.create(
    <IntlProvider {...props}>
      <MemoryRouter>
        { children }
      </MemoryRouter>
    </IntlProvider>
  );
};

export default createComponentWithIntl;