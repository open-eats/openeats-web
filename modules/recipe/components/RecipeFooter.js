import React from 'react'
import PropTypes from 'prop-types'
import { Link } from 'react-router-dom'
import {
    injectIntl,
    defineMessages,
} from 'react-intl'

const RecipeFooter = ({ slug, source, username, updateDate, showEditLink, deleteRecipe, intl }) => {
  const messages = defineMessages({
    source: {
      id: 'recipe.source',
      description: 'Source of the recipe',
      defaultMessage: 'Source'
    },
    created_by: {
      id: 'recipe.created_by',
      description: 'Created by',
      defaultMessage: 'Created by'
    },
    last_updated: {
      id: 'recipe.last_updated',
      description: 'Last Updated',
      defaultMessage: 'Last Updated'
    },
    confirm_delete: {
      id: 'recipe.confirm_delete',
      description: 'Are you sure you want to delete this recipe?',
      defaultMessage: 'Are you sure you want to delete this recipe?'
    },
  });

  let hostname = '';
  if (source) {
    // Get Host name of a URL
    let a = document.createElement('a');
    a.href = source;
    hostname = a.hostname;
  }

  const handleDelete = () => {
    if (window.confirm(intl.formatMessage(messages.confirm_delete))) {
      deleteRecipe(slug)
    }
  };

  const sourceLink = (
    <div>
      { intl.formatMessage(messages.source) }:
      <a href={ source }>{ hostname }</a>
    </div>
  );

  const editLink = (
    <Link to={ "/recipe/edit/" + slug }>
      <button className="btn btn-primary btn-sm">
        <span className="glyphicon glyphicon-pencil"/>
      </button>
    </Link>
  );

  const deleteLink = (
    <button className="btn btn-danger btn-sm" onClick={ handleDelete }>
      <span className="glyphicon glyphicon-trash"/>
    </button>
  );
  
  return (
    <div className="panel-footer">
      <div className="row">
        <div className="col-xs-6">
          { (source) ? sourceLink : null }
          <div>{ intl.formatMessage(messages.created_by) }: { username }</div>
          <div>{ intl.formatMessage(messages.last_updated) }: { updateDate }</div>
        </div>
        <div className="col-xs-6 pull-right text-right">
          { showEditLink ? editLink : null }
          { showEditLink ? deleteLink : null }
        </div>
      </div>
    </div>
  );
};

RecipeFooter.PropTypes = {
  slug: PropTypes.string.isRequired,
  source: PropTypes.string.isRequired,
  username: PropTypes.string.isRequired,
  updateDate: PropTypes.instanceOf(Date).isRequired,
  showEditLink: PropTypes.bool.isRequired,
  intl: PropTypes.object.isRequired,
};

export default injectIntl(RecipeFooter);
