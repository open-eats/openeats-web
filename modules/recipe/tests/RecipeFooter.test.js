import React from 'react';
import RecipeFooter from '../components/RecipeFooter';
import createComponentWithIntlAndRouter from '../../../config/jest/createComponentWithIntlAndRouter';

test('Test Footer without Edit Link', () => {
  const component = createComponentWithIntlAndRouter(
    <RecipeFooter
        id={ 1 }
        source={ 'google.com' }
        usernamek={ 'Ryan Noelk' }
        updateDate={ '2017-01-01' }
        showEditLink={ false }
    />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Test Footer with Edit Link', () => {
  const component = createComponentWithIntlAndRouter(
    <RecipeFooter
        id={ 1 }
        source={ 'google.com' }
        usernamek={ 'Ryan Noelk' }
        updateDate={ '2017-01-01' }
        showEditLink={ true }
    />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
