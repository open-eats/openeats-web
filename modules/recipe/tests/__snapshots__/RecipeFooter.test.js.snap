// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Footer with Edit Link 1`] = `
<div
  className="panel-footer"
>
  <div
    className="row"
  >
    <div
      className="col-xs-6"
    >
      <div>
        Source
        :
        <a
          href="google.com"
        >
          localhost
        </a>
      </div>
      <div>
        Created by
        : 
      </div>
      <div>
        Last Updated
        : 
        2017-01-01
      </div>
    </div>
    <div
      className="col-xs-6 pull-right text-right"
    >
      <a
        href="/recipe/edit/undefined"
        onClick={[Function]}
      >
        <button
          className="btn btn-primary btn-sm"
        >
          <span
            className="glyphicon glyphicon-pencil"
          />
        </button>
      </a>
      <button
        className="btn btn-danger btn-sm"
        onClick={[Function]}
      >
        <span
          className="glyphicon glyphicon-trash"
        />
      </button>
    </div>
  </div>
</div>
`;

exports[`Test Footer without Edit Link 1`] = `
<div
  className="panel-footer"
>
  <div
    className="row"
  >
    <div
      className="col-xs-6"
    >
      <div>
        Source
        :
        <a
          href="google.com"
        >
          localhost
        </a>
      </div>
      <div>
        Created by
        : 
      </div>
      <div>
        Last Updated
        : 
        2017-01-01
      </div>
    </div>
    <div
      className="col-xs-6 pull-right text-right"
    />
  </div>
</div>
`;
